### PR TITLE
config: add support for logger provider configuration

### PR DESCRIPTION
**Pull request by codeboten**

### CODE REVIEW

File: config/log.go
Comment (MrAlias) [inline config/log.go:42]: nit

```suggestion
	lp := sdklog.NewLoggerProvider(opts...)
	return lp, lp.Shutdown, nil
```

File: config/log_test.go
Comment (MrAlias) [inline config/log_test.go:313]: Missing test for simple with a valid exporter.

File: config/log.go
Comment (MrAlias) [inline config/log.go:63]:
```suggestion
	return nil, fmt.Errorf("unsupported log processor type %[1]#v (%[1]T)", processor)
```

File: config/log.go
Comment (MrAlias) [inline config/log.go:72]: Missing test case.

File: config/log.go
Comment (codeboten) [reply config/log.go:63]: did you mean to leave both v and T in here? with both it ends up logging something like:

```
"unsupported log processor type {<nil> <nil> <nil>} (config.LogRecordProcessor)"
```

File: config/log_test.go
Comment (codeboten) [reply config/log_test.go:313]: Added

File: config/log.go
Comment (codeboten) [reply config/log.go:72]: test added

File: config/log.go
Comment (MrAlias) [reply config/log.go:63]: Yeah, I was mostly thinking we need to include the type to support the rest of the message. If including the instance value as well isn't going to be helpful I'm fine removing that.

Comment (codeboten) [reply config/log.go]: i updated the error message to specify the type of processors that are expected instead. the type of processor isn't support helpful and neither are the null values

**Review by MrAlias — APPROVED** [review]

**Comment by codeboten** [comment]
@MrAlias is there anything else on this PR that's needed to move it forward?

**Comment by MrAlias** [comment]
It needs another review from a non-Splunker.

cc @open-telemetry/go-approvers

**Review by dashpole — APPROVED** [review]

**Comment by codeboten** [comment]
Thanks @MrAlias @dashpole!

**Pull request merged**
